CMakeLists.txt

```
@@ -241,6 +241,11 @@ option(USE_OBSERVERS "Use observers module." OFF)
 option(USE_OPENCL "Use OpenCL" OFF)
 option(USE_OPENCV "Use OpenCV" OFF)
 option(USE_OPENMP "Use OpenMP for parallel code" ON)
+option(USE_PRECOMPILED_HEADERS "Use pre-compiled headers to accelerate build. Requires cmake >= 3.16." OFF)
+if(USE_PRECOMPILED_HEADERS AND (CMAKE_VERSION VERSION_LESS "3.16"))
+  message(FATAL_ERROR "Precompiled headers require cmake >= 3.16")
+endif()
```

Is there a reason not to turn it on by default?
```suggestion
cmake_dependent_option(USE_PRECOMPILED_HEADERS "Use pre-compiled headers to accelerate build. Requires cmake >= 3.16." ON "CMAKE_VERSION VERSION_LESS 3.16" OFF)
```

---

I was just being conservative. There is an issue where the pch gets included everywhere, so it's possible for code to compile with PCH enabled but not without, and vice versa. It's also not necessarily a win on all machines, for example on a 32-core machine I see no benefit in overall compile times.